tests/lib/hardware/aws_ec2.py

```diff
@@ -338,9 +348,3 @@ def destroy(self):
         logger.info(f"Deleted gateway {self._gateway}")
         self._vpc.delete()
         logger.info(f"Deleted vpc {self._vpc}")
-
-    def __enter__(self):
-        return self
-
-    def __exit__(self, type, value, traceback):
-        self.destroy()
```

Comment:
don't we still need these? with then `self.destroy(skip=not settings.as_bool('_TEAR_DOWN_CLUSTER'))` ?

Reply:
They are defined in the super class.